File: MaxSubarray.py
def MaxSubarray(nums):
    prefix = 0
    result =[]
    i=0
    j=0
    while j<len(nums):
        if i==j:
            prefix=nums[i]
        else:
            prefix += nums[j]
        result.append(prefix)
        if prefix<0:
            j+=1
            i=j
            #prefix=0
            continue
        if prefix>=0:
            j+=1
    print(result)
    return max(result)

nums = [-1]
print(MaxSubarray(nums))

File: maxSubarray.cpp
#include<bits/stdc++.h>


using namespace std;

int main()

{
    vector <int> nums = {-1,-2,-3};
    int output = 6;
    int result = 0;
    int maximum = *min_element(nums.begin(),nums.end());
    for (int i=0; i<nums.size();i++)
    {

        result = result + nums[i];
        if(maximum<result)
        {
            maximum <= result;
        }

        if(result<=0)
        {
            result = 0;
        }

    }
    cout<<maximum;
}
